pxd.c

```diff
@@ -1288,18 +1288,19 @@ static int pxd_init_disk(struct pxd_device *pxd_dev)
 	blk_queue_physical_block_size(q, PXD_LBS);
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(5,19,0)
-#if defined(__EL8__)
+#if defined(__EL8__) || defined(__SUSE__)
 
-#if LINUX_VERSION_CODE != KERNEL_VERSION(5,14,0)
+#if defined(QUEUE_FLAG_DISCARD)
 	/* Enable discard support. */
 	QUEUE_FLAG_SET(QUEUE_FLAG_DISCARD,q);
-#endif
+#endif                                                       
 
-#else
+#else                                                         // #else for defined(__EL8__) || defined(__SUSE__)
```

Comment:
This doesn't need the check 
if defined(QUEUE_FLAG_DISCARD)?

Reply:
The LINUX_VERSION_CODE < KERNEL_VERSION(5,19,0) is assumed to have 'QUEUE_FLAG_DISCARD' defined.   If it does not I would rather not build than not set the discard flag and have it produce a runtime error.